src/main/java/com/creativemd/littletiles/common/tileentity/TileEntityLittleTiles.java

```diff
@@ -717,108 +724,154 @@ public IBlockState getState(AxisAlignedBB box, boolean realistic) {
                 highest = pair.value;
-            
+
         }
         return highest != null ? highest.getBlockState() : null;
     }
-    
+
     @SideOnly(Side.CLIENT)
     public boolean isRenderingEmpty() {
         return tiles.isCompletelyEmpty() && !render.hasAdditional();
     }
-    
+
     public boolean isEmpty() {
         return tiles.isCompletelyEmpty();
     }
-    
+
     @Override
     public void invalidate() {
         super.invalidate();
-        if (!preventUnload)
+        if (!preventUnload) {
             tiles.unload();
+            storageDirty = true;
+        }
     }
-    
+
     public boolean unloaded() {
         return unloaded;
     }
-    
+
     @Override
     public void onChunkUnload() {
         unloaded = true;
         super.onChunkUnload();
         tiles.unload();
+        storageDirty = true;
         if (world.isRemote) {
             tiles = null;
             render.chunkUnload();
         }
     }
-    
+
     @Override
     public void rotate(Rotation rotationIn) {
         LittleBlockTransformer.rotateTE(this, RotationUtils.getRotation(rotationIn), RotationUtils.getRotationCount(rotationIn));
         updateTiles();
     }
-    
+
     @Override
     public void mirror(Mirror mirrorIn) {
         LittleBlockTransformer.flipTE(this, RotationUtils.getMirrorAxis(mirrorIn));
         updateTiles();
     }
-    
+
     @Override
     public String toString() {
         return pos.toString();
     }
-    
+
     public void tick() {
         for (LittleStructure structure : ticking())
             structure.tick();
     }
-    
+
     public Iterable<IParentTileList> groups() {
         return tiles.groups();
     }
-    
+
     public IParentTileList noneStructureTiles() {
         return tiles;
     }
-    
+
     public Iterable<Pair<IParentTileList, LittleTile>> allTiles() {
         return tiles.allTiles();
     }
-    
+
     public IStructureTileList getStructure(int index) {
         return tiles.getStructure(index);
     }
-    
+
     public Iterable<LittleStructure> loadedStructures() {
         return tiles.loadedStructures();
     }
-    
+
     public Iterable<LittleStructure> loadedStructures(int attribute) {
         return tiles.loadedStructures(attribute);
     }
-    
+
     public Iterable<IStructureTileList> structures() {
         return tiles.structures();
     }
-    
+
     public void fillUsedIds(BitSet usedIds) {
         tiles.fillUsedIds(usedIds);
     }
-    
+
+    private IItemHandler itemHandler;
+    private boolean storageDirty = true;
+
+    protected void updateStorageState() {
+        itemHandler = null;
+        LinkedHashSet<IItemHandlerModifiable> handlers = new LinkedHashSet<>();
+        for (IStructureTileList structureHolder : structures()) {
+            try {
+                LittleStructure structure = structureHolder.getStructure();
+                if (structure.isStillAvailable() && structure instanceof LittleStorage) {
```

Comment:
Instead of checking if structure is a `LittleStorage` you should add an interface (or maybe even use an existing one), which can be implemented by structures to add their inventory.
This interface should contain the method you call later on `itemHandler()`.